add-on/src/contentScripts/ipfs-proxy/content.js

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+const browser = require('webextension-polyfill')
+const injectScript = require('./inject-script')
+
+function init () {
+  const port = browser.runtime.connect({ name: 'ipfs-proxy' })
```

Comment:
If other extensions know this name, is it possible they can connect to it? That would be bad, as they could bypass the permissions you've setup, if I understand correctly.

Reply:
It would! I'll check it out.

Reply:
...although thinking about it, the access control is all done in the extension background process, so it doesn't matter who's connecting and asking for stuff, it still has to be authorised by the user.

I'll double check but I think it might be ok.

Reply:
Ok so, the [background process is listening to `runtime.onConnect`](https://github.com/tableflip/ipfs-companion/blob/9b0009dd945c7d560a335c534d8698fd72315550/add-on/src/lib/ipfs-proxy/index.js#L67):

> Fired when a connection is made with either an extension process or a content script.
> https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/runtime/onConnect

It would need to listen to [`runtime.onConnectExternal`](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/runtime/onConnectExternal) to receive connections from other extensions.

Aside from that, even if connections from other extensions _were_ allowed, the authorization is done in the background process, so the user would still need to have approved the function to be called before the proxy server let it through to the IPFS instance.